jetty-io/src/main/java/org/eclipse/jetty/io/IncludeExcludeConnectionStatistics.java

```diff
@@ -0,0 +1,114 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.io;
+
+import java.util.AbstractSet;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.eclipse.jetty.util.IncludeExcludeSet;
+
+public class IncludeExcludeConnectionStatistics extends ConnectionStatistics
+{
+    private final IncludeExcludeSet<Class<? extends Connection>, Connection> _set = new IncludeExcludeSet<>(ConnectionSet.class);
+
+    public void include(String className) throws ClassNotFoundException
+    {
+        _set.include(connectionForName(className));
+    }
+
+    public void include(Class<? extends Connection> clazz)
+    {
+        _set.include(clazz);
+    }
+
+    public void exclude(String className) throws ClassNotFoundException
+    {
+        _set.exclude(connectionForName(className));
+    }
+
+    public void exclude(Class<? extends Connection> clazz)
+    {
+        _set.exclude(clazz);
+    }
+
+    private Class<? extends Connection> connectionForName(String className) throws ClassNotFoundException
+    {
+        Class<?> aClass = Class.forName(className);
+        if (!Connection.class.isAssignableFrom(aClass))
+            throw new IllegalArgumentException("Class is not a Connection");
+
+        @SuppressWarnings("unchecked")
+        Class<? extends Connection> connectionClass = (Class<? extends Connection>)aClass;
+        return connectionClass;
+    }
+
+    @Override
+    public void onOpened(Connection connection)
+    {
+        if (Boolean.TRUE.equals(_set.isIncludedAndNotExcluded(connection)))
+            super.onOpened(connection);
+    }
+
+    @Override
+    public void onClosed(Connection connection)
+    {
+        if (Boolean.TRUE.equals(_set.isIncludedAndNotExcluded(connection)))
+            super.onClosed(connection);
+    }
+
+    public static class ConnectionSet extends AbstractSet<Class<? extends Connection>> implements Predicate<Connection>
```

Comment:
rather than create your own set type just to provide a predicate is is probably simpler to use:
```java
new IncludeExcludeSet<>(new HashSet<>(), Connection.class::isAssignableFrom, new HashSet<>(), Connection.class::isAssignableFrom);
```

Reply:
This doesn't work, we wan't to know if assignable from one of the values present in the set, it will always be assignable from `Connection`.

To get the same logic I would need something like this
```java
HashSet<Class<? extends Connection>> includes = new HashSet<>();
HashSet<Class<? extends Connection>> excludes = new HashSet<>();

_set = new IncludeExcludeSet<>(includes, connection ->
{
    if (connection == null)
        return false;
    return includes.stream().anyMatch(c -> c.isAssignableFrom(connection.getClass()));
},
excludes, connection ->
{
    if (connection == null)
        return false;
    return excludes.stream().anyMatch(c -> c.isAssignableFrom(connection.getClass()));
});
```
so seems easier to just define the set type.